rest/blip_api_test.go

```diff
@@ -4532,3 +4532,109 @@ func TestBlipRevokeNonExistentRole(t *testing.T) {
 		return rt.GetDatabase().DbStats.CBLReplicationPull().NumPullReplCaughtUp.Value()
 	}, 1)
 }
+
+func TestBlipGetCollections(t *testing.T) {
+
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyAll)
+
+	rt := NewRestTester(t, &RestTesterConfig{
+		guestEnabled: true,
+		DatabaseConfig: &DatabaseConfig{
+			DbConfig: DbConfig{
+				Scopes: ScopesConfig{
+					"fooScope": ScopeConfig{
+						Collections: map[string]CollectionConfig{
+							"fooCollection": {},
+						},
+					},
+				},
+			},
+		},
+	})
+	defer rt.Close()
+
+	btc, err := NewBlipTesterClientOptsWithRT(t, rt, nil)
+	require.NoError(t, err)
+	defer btc.Close()
+
+	checkpointBody := db.Body{}
+	checkpointID1 := "checkpoint1"
+	dbInstance := db.Database{DatabaseContext: rt.GetDatabase()}
+	//TODO, use db.checkpointDocIDPrefix exported ?
+	revID, err := dbInstance.PutSpecial(db.DocTypeLocal, "checkpoint/"+checkpointID1, checkpointBody)
+	require.NoError(t, err)
+	checkpoint1RevID := "0-1"
+	require.Equal(t, checkpoint1RevID, revID)
+	testCases := []struct {
+		requestBody db.GetCollectionsRequestBody
+		resultBody  []db.Body
+		errorCode   string
+	}{
+		{
+			requestBody: db.GetCollectionsRequestBody{
+				CheckpointIDs: []string{"id"},
+				Collections:   []string{"_default._default"},
+			},
+			resultBody: []db.Body{nil},
+			errorCode:  "",
+		},
+		{
+			requestBody: db.GetCollectionsRequestBody{
+				CheckpointIDs: []string{"id", "id2"},
+				Collections:   []string{"_default._default"},
+			},
+			resultBody: []db.Body{nil},
+			errorCode:  fmt.Sprintf("%d", http.StatusBadRequest),
+		},
```

Comment:
It might be helpful to include a short test name that describes what's being tested for each case (which can also be passed into `t.Run(...)`)

This one took a second to figure out that the expected error was due to the mismatched array lengths.